src/core/transaction.c

```diff
@@ -216,19 +216,20 @@ static int delete_one_unmergeable_job(Transaction *tr, Job *job) {
         return -EINVAL;
 }
 
-static int transaction_merge_jobs(Transaction *tr, sd_bus_error *e) {
+static int transaction_ensure_mergable(Transaction *tr, bool matters_to_anchor, sd_bus_error *e) {
         Job *j;
         int r;
 
         assert(tr);
 
-        /* First step, check whether any of the jobs for one specific
-         * task conflict. If so, try to drop one of them. */
         HASHMAP_FOREACH(j, tr->jobs) {
                 JobType t;
 
                 t = j->type;
                 LIST_FOREACH(transaction, k, j->transaction_next) {
+                        if (j->matters_to_anchor || k->matters_to_anchor != matters_to_anchor)
```

Comment:
Hmm? This means all units that matter to anchor are exempted from validation/deletion, which might trigger the assertion on job_type_merge_and_collapse() when actually merging below, no?

Reply:
No, we anyway check all pairs of jobs. Hence the assertion does not triggered. This just changes the order of the check.